packages/editor-ui/src/mixins/completionManager.ts

```diff
@@ -0,0 +1,76 @@
+import mixins from 'vue-typed-mixins';
+import { ExpressionExtensions } from 'n8n-workflow';
+import { EditorView, ViewUpdate } from '@codemirror/view';
+
+import { expressionManager } from './expressionManager';
+
+export const completionManager = mixins(expressionManager).extend({
+	data() {
+		return {
+			editor: {} as EditorView,
+		};
+	},
+	computed: {
+		expressionExtensionsCategories() {
+			return ExpressionExtensions.reduce<Record<string, string | undefined>>((acc, cur) => {
+				for (const fnName of Object.keys(cur.functions)) {
+					acc[fnName] = cur.typeName;
+				}
+
+				return acc;
+			}, {});
+		},
+	},
+	methods: {
+		trackCompletion(viewUpdate: ViewUpdate, parameterPath: string) {
```

Comment:
does this really need to be a mixin.. goal of this mixin is this one function used in one component.. as we move towards vue3, we want to get rid of mixins.. try compasables next time..

Reply:
This is used twice in `ExpressionEditorModalInput` and `InlineExpressionEditorInput` and when we refactor `CodeNodeEditor` it will be reused there as well.